capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -29,57 +30,20 @@ export class WikiMediaComponent implements OnInit, OnDestroy {
   }
 
   getResults(queryString: string) {
-    // TODO(sarahhud): Add error handling for wikipedia request and testing.
     this.wikiService.search(queryString)
       .pipe(takeUntil(this.destroy$))
```

Comment:
I could be completely wrong here and definitely want a second opinion from Matt/Nicholas as my understanding of rxjs is pretty basic. But, I don't think this is a situation when you need destroy$ and takeuntil(this.destroy$). 1) I don't believe this component is ever completely destroyed and even if it is 2) The unsubscribing from a single API call is relatively basic as it returns a single response of the whole article in a json

Reply:
A single API call using httpClient I think unsubscribes on its own. One lightweight way we can force it to unsubscribe just in case though is to use "take(1)" instead of "takeUntil(this.destroyed$)". This will unsubscribe after the first emission, which is typically what we want for an API call, instead of waiting until the component itself is destroyed.

Reply:
https://angular.io/tutorial/toh-pt6#httpclient-methods-return-one-value
"An observable from HttpClient always emits a single value and then completes, never to emit again."
If we want to trust the behavior of HttpClient, then take(1) might be redundant also. I can go either way.

Reply:
Slightly confused after reading all of your comments. Should I just entirely remove the take until? @NicholasCobbFromGoogle  @mbehrlich

Reply:
Yes, definitely remove takeUntil. If you really, really, really want to be super safe, you can add take(1) instead, but you don't really need that either.